File: 0167-Two Sum II - Input array is sorted/0167-Two Sum II - Input array is sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int start = 0, end = numbers.size() - 1;
        vector<int> result(2);
        while (start < end)
        {
            int sum = numbers[start] + numbers[end];
            if (sum == target)
            {
                result[0] = start + 1;
                result[1] = end + 1;
                break;
            }
            else if (sum < target)
            {
                ++start;
            }
            else
            {
                --end;
            }
        }
        
        return result;
    }
};

File: 0101-0200/0167-Two Sum II - Input array is sorted/0167-Two Sum II - Input array is sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        start = 0
        end = len(numbers) - 1
        result = [0] * 2
        while start < end:
            curr = numbers[start] + numbers[end]
            if curr < target:
                start += 1
            elif curr > target:
                end -= 1
            else:
                result[0] = start + 1
                result[1] = end + 1
                break
        return result
